programs/locker/src/instructions/root_escrow_instructions/create_vesting_escrow_from_root.rs

```diff
@@ -0,0 +1,197 @@
+use crate::*;
+use anchor_spl::associated_token::AssociatedToken;
+use anchor_spl::token_interface::{Mint, TokenAccount, TokenInterface};
+use merkle_verify::verify;
+use solana_program::hash::hashv;
+use util::{
+    calculate_transfer_fee_included_amount, parse_remaining_accounts, transfer_from_root_escrow,
+    AccountsType, ParsedRemainingAccounts,
+};
+
+const LEAF_PREFIX: &[u8] = &[0];
+
+#[derive(AnchorSerialize, AnchorDeserialize)]
+pub struct CreateVestingEscrowFromRootParams {
+    pub vesting_start_time: u64,
+    pub cliff_time: u64,
+    pub frequency: u64,
+    pub cliff_unlock_amount: u64,
+    pub amount_per_period: u64,
+    pub number_of_period: u64,
+    pub update_recipient_mode: u8,
+    pub cancel_mode: u8,
+}
+
+impl CreateVestingEscrowFromRootParams {
+    pub fn into_vesting_escrow_params(&self) -> CreateVestingEscrowParameters {
+        CreateVestingEscrowParameters {
+            vesting_start_time: self.vesting_start_time,
+            cliff_time: self.cliff_time,
+            frequency: self.frequency,
+            cliff_unlock_amount: self.cliff_unlock_amount,
+            amount_per_period: self.amount_per_period,
+            number_of_period: self.number_of_period,
+            update_recipient_mode: self.update_recipient_mode,
+            cancel_mode: self.cancel_mode,
+        }
+    }
+}
+
+/// Accounts for [locker::create_vesting_escrow_from_root].
+#[event_cpi]
+#[derive(Accounts)]
+pub struct CreateVestingEscrowFromRootCtx<'info> {
+    /// Root Escrow.
+    #[account(
+        mut,
+        has_one = token_mint,
+    )]
+    pub root_escrow: AccountLoader<'info, RootEscrow>,
+
+    /// CHECK: Base.
+    #[account(
+        seeds = [
+            b"base",
+            root_escrow.key().as_ref(),
+            recipient.key().as_ref(), // unique
+        ],
+        bump,
+    )]
+    pub base: UncheckedAccount<'info>,
```

Comment:
Should be renamed to avoid confusion with the `base` signer in `create_root_escrow`